### ISSUE TITLE
Create images/diagrams to explain main message flows and usecases

### ISSUE THREAD

**Issue by sanjeewa-malalgoda** [theagentrouter/agent-router#15]
Project requires clear and visually engaging images/diagrams to explain its core message flows and use cases. These visuals will help developers, and end-users better understand the architecture, capabilities, and practical implementations of the gateway.

Requirements
Message Flows:
Illustrate the main data flow, including:
- Request ingress through the gateway.
- Routing to LLM backends or API backends.
- Response handling and any transformations (e.g., prompt decorating or semantic caching).
- Include dynamic features like weighted routing and multi-LLM backend switching.

Use Cases:
Multi-LLM Backend Support
Prompt Templating and Decorating
Semantic Caching
AI API Observability

**Comment by sanjeewa-malalgoda** [comment]
<img width="1500" alt="Screenshot 2024-12-02 at 07 06 27" src="https://github.com/user-attachments/assets/ce37b1cc-8df5-4e82-ab88-f9d8c1545995">
<img width="1177" alt="Screenshot 2024-12-02 at 07 06 10" src="https://github.com/user-attachments/assets/4804dff2-1372-4295-a734-98814f9767c0">
<img width="887" alt="Screenshot 2024-12-02 at 06 36 06" src="https://github.com/user-attachments/assets/8981f0b0-60c3-49a7-af9d-bef3d0880b1d">

**Comment by sanjeewa-malalgoda** [comment]
[diagram-source-files.zip](https://github.com/user-attachments/files/17971455/diagram-source-files.zip)

**Comment by missBerg** [comment]
Wondering if we should make the difference between the Control Plane, Data Plane, and External Processes clear?
As in Envoy Proxy being the data plane, and specific external processes extend functionality.
And that the Gateway control plane configures Envoy Proxy and External Processes?

CC @mathetake

**Comment by mathetake** [comment]
yeah the relationship with Envoy Gateway and where External Processing sits in the request flow and how it's managed might be a good content in the diagram. This should be in line with https://github.com/envoyproxy/ai-gateway/issues/16 as well (which one of the maintainers will work on it for the next couple of weeks)

**Comment by sanjeewa-malalgoda** [comment]
Thanks for suggestions. Will those to diagrams and i should be able to get some idea from #16 too.

**Comment by mathetake** [comment]
so @yuzisun is working on the design doc, and after he finishes it let's get back to this and polish the diagram

**Comment by mathetake** [comment]
I _believe_ this is superseded by #52
